### DISCUSSION TITLE
alwaysAllowTs is ignored as attempting to migrate in bun

### DISCUSSION THREAD

**Discussion by ghost** [mikro-orm/mikro-orm#5052] [Q&A] [answered]
### Describe the bug

Migration fails in bun.
Bun has its own typescript module resolution system, hence using ts-node is not the expectation of bun.

### Reproduction

[package.json]
`"mikro-orm": {
    "useTsNode": false, // This is additional, even if I remove this, the error doesn't change 
    "alwaysAllowTs": true,
    "configPaths": [
      "./src/mikro-orm.config.ts",
      "./dist/mikro-otm.config.js"
    ]
  }`

[bunx mikro-orm debug]
Current MikroORM CLI configuration
 - dependencies:
   - mikro-orm 5.9.6
   - node 18.17.1
   - typescript 5.3.3
 - package.json found
 - searched config paths:
   - /Users/.../Desktop/Git/test-mikro/src/mikro-orm.config.ts (found)
   - /Users/.../Desktop/Git/test-mikro/dist/mikro-otm.config.js (not found)
   - /Users/.../Desktop/Git/test-mikro/mikro-orm.config.ts (found)
   - /Users/.../Desktop/Git/test-mikro/src/mikro-orm.config.js (not found)
   - /Users/.../Desktop/Git/test-mikro/mikro-orm.config.js (not found)
- configuration not found (Unknown file extension ".ts" for /Users/.../Desktop/Git/test-mikro/src/mikro-orm.config.ts)

[tsconfig.json]
{
  "compilerOptions": {
    "lib": ["ESNext"],
    "module": "esnext",
    "target": "esnext",
    "moduleResolution": "bundler",
    "moduleDetection": "force",
    "allowImportingTsExtensions": true,
    "experimentalDecorators": true,
    "emitDecoratorMetadata": true,
    "strictPropertyInitialization": false,
    "esModuleInterop": true,
    "noEmit": true,
    "composite": true,
    "strict": true,
    "downlevelIteration": true,
    "skipLibCheck": true,
    "jsx": "react-jsx",
    "allowSyntheticDefaultImports": true,
    "forceConsistentCasingInFileNames": true,
    "allowJs": true,
    "types": [
      "bun-types" // add Bun global
    ]
  }
}

### What driver are you using?

None

### MikroORM version

5.9

### Node.js version

bun:1.0.18

### Operating system

macos

### Validations

- [X] Read the [Contributing Guidelines](https://github.com/mikro-orm/mikro-orm/blob/master/CONTRIBUTING.md).
- [X] Read the [docs](https://mikro-orm.io/docs).
- [X] Check that there isn't [already an issue](https://github.com/mikro-orm/mikro-orm/issues) that reports the same bug to avoid creating a duplicate.
- [X] Check that this is a concrete bug. For Q&A open a [GitHub Discussion](https://github.com/mikro-orm/mikro-orm/discussions) or join our [Slack](https://join.slack.com/t/mikroorm/shared_invite/enQtNTM1ODYzMzM4MDk3LWM4ZDExMjU5ZDhmNjA2MmM3MWMwZmExNjhhNDdiYTMwNWM0MGY5ZTE3ZjkyZTMzOWExNDgyYmMzNDE1NDI5NjA).
- [X] The provided reproduction is a [minimal reproducible example](https://stackoverflow.com/help/minimal-reproducible-example) of the bug.

**Comment by B4nan** [comment]
`alwaysAllowTs` flag works fine, what it does is allowing TS files, which clearly works as you can see them in the "searched config paths" - that is what the flag controls, it won't add TS support on its own, that is up to you to do - by running the CLI script via `bun` (or tsx, ts-node, whatever you wish to use).

MikroORM won't register anything else then ts-node automatically.

**Reply by ghost** [reply, marked as answer]
Yes. There was no issue. I found the solution. 
`bun --bun mikro-orm migration:create -i`
NOT `bun mikro-orm migration:create -i.`
